stan/math/prim/fun/gaus_interp.hpp

```diff
@@ -0,0 +1,198 @@
+#ifndef STAN_MATH_PRIM_FUN_GAUS_INTERP
+#define STAN_MATH_PRIM_FUN_GAUS_INTERP
+
+#include <cmath>
+#include <vector>
+#include <algorithm>
+#include <stan/math/prim/fun/conv_gaus_line.hpp>
+
+using namespace std;
+
+namespace stan {
+namespace math {
+
+// struct for passing precomputation data
+struct gaus_interp_params {
+  std::vector<double> as;
+  std::vector<double> bs;
+  double sig2;
+};
+
+namespace internal {
+
+/*
+find the smallest difference between successive elements in a sorted vector
+*/
+template <typename Tx>
+double min_diff(int n, std::vector<Tx> const& xs) {
+  double dmin = value_of(xs[1]) - value_of(xs[0]);
+  for (int i = 1; i < n - 1; i++) {
+    if (value_of(xs[i + 1]) - value_of(xs[i]) < dmin) {
+      dmin = value_of(xs[i + 1]) - value_of(xs[i]);
+    }
+  }
+  return dmin;
+}
```

Comment:
Do you want the smallest absolute difference (two closest number)?

Reply:
This will only be used on vectors with increasing elements, so the differences here should all be positive.